modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -274,27 +225,108 @@ public void testFreeMarkerNotification() throws Exception {
 					}
 				},
 				group.getGroupKey());
-		}
-		finally {
-			PermissionThreadLocal.setPermissionChecker(
-				originalPermissionChecker);
-			PrincipalThreadLocal.setName(originalName);
-		}
 
-		termValues = _getFreeMarkerTermValues(
-			childObjectDefinition,
-			_objectEntryLocalService.getPersistedModel(objectEntry.getId()),
-			_getTermNames(
-				childObjectDefinition.getObjectDefinitionId(),
-				SetUtil.fromArray(
-					"Basic Information",
-					childObjectDefinition.getLabel(
-						LanguageUtil.getLanguageId(LocaleUtil.US)))));
+			Map<String, Object> termValues = _getFreeMarkerTermValues(
+				notificationTemplateId, childObjectDefinition,
+				_objectEntryLocalService.getPersistedModel(objectEntry.getId()),
+				_getTermNames(
+					childObjectDefinition.getObjectDefinitionId(),
+					SetUtil.fromArray(
+						"Basic Information",
+						childObjectDefinition.getLabel(
+							LanguageUtil.getLanguageId(LocaleUtil.US)))));
+
+			String body = StringUtil.merge(
+				termValues.keySet(), StringPool.POUND);
+
+			NotificationTemplate notificationTemplate =
+				notificationTemplateLocalService.getNotificationTemplate(
+					notificationTemplateId);
+
+			notificationTemplate.setBody(body);
+
+			notificationTemplateLocalService.updateNotificationTemplate(
```

Comment:
Since probably we won't need the `notificationTemplateId` as a parameter of  `_getFreeMarkerTermValues`, we won't need to update the `NotificationTemplate` anymore. So, build the `body` before creating the notification template.

Reply:
Done!